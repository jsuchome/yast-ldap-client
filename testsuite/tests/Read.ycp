/**
 *  Read.ycp
 *  Test of Ldap:Read function
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    include "testsuite.ycp";
    // testedfiles: Ldap.ycp

    map READ = $[
	"etc": $[
	    "nsswitch_conf" : $[
		"passwd" : "compat",
		"group"  : "compat",
		"passwd_compat": "ldap",
		"group_compat": "ldap",
	    ],
	    "ldap_conf" : $[
		"v" : $[
		    "/etc/ldap.conf": $[
			"host" : "localhost",
			"base" : "dc=suse,dc=cz",
			"nss_base_passwd" : nil,
			"nss_base_shadow" : nil,
			"nss_base_group" : nil,
			"ldap_version": nil,
			"ssl": nil,
			"pam_password": "crypt",
		    ]
		]
	    ],
	    // /etc/security/pam_*
	    "security": $[
		"section": $[
		    "/etc/security/pam_unix2.conf": $[],
		],
		"v": $[
		    "/etc/security/pam_unix2.conf": $[
			"auth"	: ""
		    ],
		]
	    ],
	],
	"sysconfig": $[
	    "ldap": $[
		"BASE_CONFIG_DN": nil,
		"BIND_DN": "uid=manager,dc=suse,dc=cz",
		"FILE_SERVER": "no"
	    ]
	],
	"init": $[
	    "scripts": $[
		"exists": false,
	    ],
	],
	"passwd": $[
	    "passwd": $[
		"plusline": "+",
		"pluslines": ["+"]
	    ]
	],
	"product" : $[ "features" :
	    $[ "EVMS_CONFIG" : "nazdar" ],
	],
    ];

    map EX = $[
	"target": $[
	    "bash": 0,
	],
	"passwd": $[
	    "init": true
	]
    ];
    TESTSUITE_INIT([READ,$[],$[]], nil);

    import "Ldap";

    DUMP ("==== reading... ============================");

    TEST(``(Ldap::Read ()), [ READ, $[], EX ], 0);

    DUMP ("============================================");

    DUMP ( sformat ("ldap used: -%1-", Ldap::start) );

    DUMP ( sformat ("nsswitch: -%1-", Ldap::nsswitch) );

    DUMP ( sformat ("base config DN: -%1-", Ldap::base_config_dn) );

    DUMP ( sformat ("bind DN: -%1-", Ldap::bind_dn) );
}
