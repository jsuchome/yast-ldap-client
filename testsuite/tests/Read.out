Dump	==== reading... ============================
Execute	.target.bash "/usr/bin/grep '^passwd:.*ldap' /etc/nsswitch.conf" 0
Read	.etc.ldap_conf.v."/etc/ldap.conf"."host" "localhost"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."base" "dc=suse,dc=cz"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."ldap_version" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."ssl" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_passwd" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_shadow" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_group" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."pam_password" "crypt"
Read	.etc.nsswitch_conf.passwd "compat ldap"
Read	.etc.nsswitch_conf.group "compat ldap"
Read	.sysconfig.ldap.BASE_CONFIG_DN nil
Read	.sysconfig.ldap.FILE_SERVER "no"
Read	.sysconfig.ldap.BIND_DN "uid=manager,dc=suse,dc=cz"
Read	.sysconfig.displaymanager.DISPLAYMANAGER 0
Return	true
Dump	============================================
Dump	ldap used: -true-
Dump	nsswitch: -$["group":["compat", "ldap"], "passwd":["compat", "ldap"]]-
Dump	base config DN: --
Dump	bind DN: -uid=manager,dc=suse,dc=cz-
